tests/test_scylla.c

```diff
@@ -149,36 +149,74 @@ void test_scylla(void) {
   db_client_service_free(&db_client_service);
 }
 
-void test_db_identity_table(void) {
-  db_client_service_t db_client_service;
-
-  const char hashes[][NUM_FLEX_TRITS_HASH] = {
-      {TRANSACTION_1}, {TRANSACTION_1}, {TRANSACTION_2}, {TRANSACTION_2}, {TRANSACTION_3}, {TRANSACTION_3},
-  };
+void test_db_get_identity_objs_by_status(db_client_service_t* db_client_service) {
   const char expected_select_hashes[][NUM_FLEX_TRITS_HASH] = {
       {TRANSACTION_2}, {TRANSACTION_2}, {TRANSACTION_1}, {TRANSACTION_1}, {TRANSACTION_3}, {TRANSACTION_3},
   };
-  db_client_service.host = strdup(host);
-  TEST_ASSERT_EQUAL_INT(db_client_service_init(&db_client_service), SC_OK);
-  TEST_ASSERT_EQUAL_INT(db_init_identity_keyspace(&db_client_service, true, keyspace_name), SC_OK);
-  for (int i = 0; i < 6; i++) {
-    db_insert_tx_into_identity(&db_client_service, i, hashes[i], i / 2);
+  db_identity_array_t* db_identity_array = db_identity_array_new();
+  db_get_identity_objs_by_status(db_client_service, INSERTING_TXN, db_identity_array);
+  db_get_identity_objs_by_status(db_client_service, PENDING_TXN, db_identity_array);
+  db_get_identity_objs_by_status(db_client_service, CONFIRMED_TXN, db_identity_array);
+  db_identity_t* itr;
+  int idx = 0;
+  IDENTITY_TABLE_ARRAY_FOREACH(db_identity_array, itr) {
+    TEST_ASSERT_EQUAL_MEMORY(db_ret_identity_hash(itr), (flex_trit_t*)expected_select_hashes[idx % 6],
+                             sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+    idx++;
   }
+  db_identity_array_free(&db_identity_array);
+}
 
+void test_db_get_identity_objs_by_id(db_client_service_t* db_client_service) {
+  const char expected_select_hashes[][NUM_FLEX_TRITS_HASH] = {{TRANSACTION_2}, {TRANSACTION_3}, {TRANSACTION_1}};
   db_identity_array_t* db_identity_array = db_identity_array_new();
-  db_select_identity_table(&db_client_service, INSERTING_TXN, db_identity_array);
-  db_select_identity_table(&db_client_service, PENDING_TXN, db_identity_array);
-  db_select_identity_table(&db_client_service, CONFIRMED_TXN, db_identity_array);
+  db_get_identity_objs_by_id(db_client_service, 3, db_identity_array);
```

Comment:
Maybe use macro to replace using integer directly

Reply:
OK!